### ISSUE TITLE
Parallelise note commitment tree point conversions

### ISSUE THREAD

**Issue by teor2345** [ZcashFoundation/zebra#5425]
## Motivation

We can speed up note commitment tree updates by running point conversions in parallel with tree updates.

### Designs

Run the following operations in parallel rayon iterators or threads:
- each point conversion, or batches of point conversions, in any order, in multiple threads (`cm_x.into()`)
- appending notes, in blockchain order, in a single thread (`self.inner.append(...)`)

We can start appending notes as soon as the first point conversion is completed.

This change can be made for Sprout, Sapling, and Orchard. Sprout is a lower priority because it's not currently being used much.

Orchard example:
https://github.com/ZcashFoundation/zebra/blob/267db7f956c741c5f4369722eb9aaa0c49c359ac/zebra-chain/src/orchard/tree.rs#L309-L310

`zcashd` example of out of order note insertion using `shardtree`:
https://github.com/zcash/incrementalmerkletree/pull/50

**Comment by teor2345** [comment]
This is currently optional. We've talked about it a few times, so I wanted to write it down so we had a record.

But I think we can close it, unless we decide we need it to fix #5420, and that #5420 is an important enough fix to schedule soon.

Edit: or we decide we want to fix #5709, which we need to fix before this ticket will help with sync speed.
